csharp/extractor/Semmle.Extraction.CSharp/Entities/Property.cs

```diff
@@ -69,12 +72,24 @@ public override void Populate()
                     Context.PopulateLater(() => Expression.Create(Context, expressionBody, this, 0));
                 }
 
+                int child = 1;
                 foreach (var initializer in declSyntaxReferences.
                     Select(n => n.Initializer).
-                    Where(i => i != null).
-                    Select(i => i.Value))
+                    Where(i => i != null))
                 {
-                    Context.PopulateLater(() => Expression.Create(Context, initializer, this, 1));
+                    Context.PopulateLater(() =>
+                    {
+                        var loc = Context.Create(initializer.GetLocation());
+                        var annotatedType = new AnnotatedType(type, TypeAnnotation.None);
+                        var simpleAssignExpr = new Expression(new ExpressionInfo(Context, annotatedType, loc, ExprKind.SIMPLE_ASSIGN, this, child++, false, null));
+                        Expression.CreateFromNode(new ExpressionNodeInfo(Context, initializer.Value, simpleAssignExpr, 0));
+                        var access = new Expression(new ExpressionInfo(Context, annotatedType, Location, ExprKind.PROPERTY_ACCESS, simpleAssignExpr, 1, false, null));
+                        Context.Emit(Tuples.expr_access(access, this));
+                        if (!symbol.IsStatic)
```

Comment:
Should there also be an `expr_call` here to the setter?

Reply:
We don't extract `expr_call` for properties, but instead do it in QL: https://github.com/Semmle/ql/blob/master/csharp/ql/src/semmle/code/csharp/exprs/Call.qll#L584.